src/HttpRequest.cc

```diff
@@ -471,9 +471,10 @@ HttpRequest::packFirstLineInto(Packable * p, bool full_uri) const
     const SBuf tmp(full_uri ? effectiveRequestUri() : url.originForm());
 
     // form HTTP request-line
-    p->appendf(SQUIDSBUFPH " " SQUIDSBUFPH " HTTP/%d.%d\r\n",
+    p->appendf(SQUIDSBUFPH " " SQUIDSBUFPH " %s/%d.%d\r\n",
                SQUIDSBUFPRINT(method.image()),
                SQUIDSBUFPRINT(tmp),
+               AnyP::ProtocolType_str[http_ver.protocol],
```

Comment:
FTR: previous bug in ICY and ICAP error messages %W output. Now fixed to ensure identical %W/%R logic.